src/read/mod.rs

```diff
@@ -307,7 +341,7 @@ where
         Map: FnMut(&K, &Values<V, S>) -> Target,
         Collector: FromIterator<Target>,
     {
-        Collector::from_iter(self.read().iter().map(|(k, v)| f(k, v)))
+        Collector::from_iter(self.read().iter().flatten().map(|(k, v)| f(k, v)))
```

Comment:
What purpose does this change serve?

Reply:
`read()` now returns an Option - this gets rid of it.

Reply:
Ah, right, because `Option::iter` iterates over the one element in the `Option`. That's neat I suppose :p